### PR TITLE
Autocorrect unenclosed strings and lists

### PR DESCRIPTION
**Pull request by mc-oofert**
# Description

If the user tries to change a variable to stuff like
list(hi
"im a string
it automatically corrects to
list(hi)
"im a string"
# Type of change

- [x] Minor changes or tweaks (quality of life stuff)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: internal/app/ui/cpvareditor/vareditor.go [hunk 1]
```diff
@@ -210,6 +220,29 @@ func (v *VarEditor) isCurrentVarInitial(varName string) bool {
...
+		variable_bad = !strings.HasSuffix(varValue, enclosing_c)
+	}
+
+	if variable_bad {
+		dialog.Open(dialog.TypeInformation{
+			Title:       "Variable Error!",
+			Information: "Variable " + varName + ", of value\n" + varValue + "\nis not enclosed with a " + enclosing_c + " !",
+		})
+	}
+	return variable_bad
```
Comment (SpaiR) [inline internal/app/ui/cpvareditor/vareditor.go:243]: In Go we use `camelCase` style to name vars. Please address it.

File: internal/app/ui/cpvareditor/vareditor.go [hunk 2]
```diff
@@ -210,6 +220,29 @@ func (v *VarEditor) isCurrentVarInitial(varName string) bool {
...
+	} else if is_list {
+		//if list is enclosed
+		enclosing_c = `)`
+		variable_bad = !strings.HasSuffix(varValue, enclosing_c)
+	}
+
+	if variable_bad {
+		dialog.Open(dialog.TypeInformation{
+			Title:       "Variable Error!",
+			Information: "Variable " + varName + ", of value\n" + varValue + "\nis not enclosed with a " + enclosing_c + " !",
```
Comment (SpaiR) [inline internal/app/ui/cpvareditor/vareditor.go:240]: Consider to use https://gobyexample.com/string-formatting

**Review by SpaiR — CHANGES_REQUESTED** [review]

**Comment by SpaiR** [comment]
As a thought: maybe it's better to auto-correct a mistake?

**Comment by mc-oofert** [comment]
> As a thought: maybe it's better to auto-correct a mistake?

gonna try that once i finish up on the screenshot pr

**Comment by mc-oofert** [comment]
> As a thought: maybe it's better to auto-correct a mistake?

did

File: internal/app/ui/cpvareditor/vareditor.go (hunk 1 above)
Comment (mc-oofert) [reply internal/app/ui/cpvareditor/vareditor.go]: addressed

File: internal/app/ui/cpvareditor/vareditor.go (hunk 2 above)
Comment (mc-oofert) [reply internal/app/ui/cpvareditor/vareditor.go]: irrelevant because i just removed the dialog

**Review by SpaiR — APPROVED** [review]

**Pull request merged**